lib/pure/math.nim

```diff
@@ -980,27 +985,40 @@ func euclMod*[T: SomeNumber](x, y: T): T {.since: (1, 5, 1).} =
   if result < 0:
     result += abs(y)
 
-when not defined(js):
-  func c_frexp*(x: float32, exponent: var int32): float32 {.
-      importc: "frexp", header: "<math.h>".}
-  func c_frexp*(x: float64, exponent: var int32): float64 {.
-      importc: "frexp", header: "<math.h>".}
-  func frexp*[T, U](x: T, exponent: var U): T =
-    ## Split a number into mantissa and exponent.
-    ##
-    ## ``frexp`` calculates the mantissa m (a float greater than or equal to 0.5
-    ## and less than 1) and the integer value n such that ``x`` (the original
-    ## float value) equals ``m * 2**n``. frexp stores n in `exponent` and returns
-    ## m.
-    ##
-    runnableExamples:
-      var x: int
-      doAssert frexp(5.0, x) == 0.625
-      doAssert x == 3
-    var exp: int32
+func frexp*[T: float32|float64](x: T, exponent: var int): T =
```

Comment:
since you're fixing it, might as well improve it further. current signature is bad because exponent is write-only. how about adding this:

```nim
func frexp*[T: float32|float64](x: T): tuple[frac: T, exp: int]
```

(and making the pre-existing one call this to avoid duplication)

Reply:
sure